### DISCUSSION TITLE
Getting warning during estimating random sampling

### DISCUSSION THREAD

**Discussion by aashhishh** [moj-analytical-services/splink#762] [Q&A]
Hello, I am pretty new to the De Duplication and all the stuff, please can you help me understand this why I am getting this error?
![image](https://user-images.githubusercontent.com/94435490/189137085-6c0a360c-b96c-4564-b529-028329225e2a.png)
and also why some certain columns are not trained for m values?
![image](https://user-images.githubusercontent.com/94435490/189137536-4c772e75-8ca4-461d-befc-7c62c29ab955.png)
![image](https://user-images.githubusercontent.com/94435490/189137697-be43a14f-d377-4652-9593-f4eca3dd198c.png)

**Comment by RobinL** [comment]
How big is the dataset?  For training u values, you'd usually want to set the target rows much higher than 1500

**Reply by aashhishh** [reply]
Dataset consists of 2000rows, and I did change the values of target rows to 10M, still getting these warnings.
After changing some settings, now facing new error.
![image](https://user-images.githubusercontent.com/94435490/189814615-cb44935e-ac8c-41df-a84d-88e9496c306d.png)
Below is the settings attached, after changing some comparison operations.
![image](https://user-images.githubusercontent.com/94435490/189814638-d1b74183-11e6-491c-860a-94ff5fc703f5.png)

**Reply by RobinL** [reply]
I think a lot of these issues are occurring because you have a very small input dataset.

The warnings are not necessarily a problem - it depends whether they are true!  For example, are there at least two records with the same drivers licence number of data of birth?  

The model will still work in these circumstances.

The 'duplicate names' problem can probably be solved by restarting the kernel and running everything from scratch.

In order to help more, we'd probably need to see your full code and at least a couple of example lines of data (which you can fake, if the real data is sensitive)

**Reply by aashhishh** [reply]
Yes, there are at least 20-30 rows which are repeated for 2-3 times. 
I am sharing my whole notebook for your reference.
[splink_pyspark.zip](https://github.com/moj-analytical-services/splink/files/9556201/splink_pyspark.zip)

**Reply by aashhishh** [reply]
There's another error which I am facing for another dataset. 
Instead of unique_id column, my dataset have UID. Can't we just define somewhere that UID is unique_id column?
![image](https://user-images.githubusercontent.com/94435490/190068068-c180b8f5-1351-486d-95e2-8d4e62a81625.png)

**Reply by aashhishh** [reply]
![image](https://user-images.githubusercontent.com/94435490/190068394-a57b3a5d-3337-4fa3-8e3a-f3f5f4b2a210.png)

**Reply by RobinL** [reply]
See the `unique_id_column_name` setting
https://moj-analytical-services.github.io/splink/settings_dict_guide.html#unique_id_column_name

**Reply by aashhishh** [reply]
Appreciated for the link. Can you please also look at the code I have shared.
